charts/connectors/values.go

```diff
@@ -0,0 +1,242 @@
+		Offset string `json:"offset"`
+		Config string `json:"config"`
+		Status string `json:"status"`
+	} `json:"topic"`
+}
+
+type TLS struct {
+	Enabled bool `json:"enabled"`
+	CA      struct {
+		SecretRef           string `json:"secretRef"`
+		SecretNameOverwrite string `json:"secretNameOverwrite"`
+	} `json:"ca"`
+	Cert struct {
+		SecretRef           string `json:"secretRef"`
+		SecretNameOverwrite string `json:"secretNameOverwrite"`
+	} `json:"cert"`
+	Key struct {
+		SecretRef           string `json:"secretRef"`
+		SecretNameOverwrite string `json:"secretNameOverwrite"`
+	} `json:"key"`
+}
+
+type Auth struct {
+	SASL struct {
+		Enabled   bool   `json:"enabled"`
+		Mechanism string `json:"mechanism"`
+		SecretRef string `json:"secretRef"`
+		UserName  string `json:"userName"`
+	} `json:"sasl"`
+}
+
+type Logging struct {
+	Level string `json:"level"`
+}
+
+type Monitoring struct {
+	Enabled           bool              `json:"enabled"`
+	ScrapeInterval    Duration          `json:"scrapeInterval"`
+	Labels            map[string]string `json:"labels"`
+	Annotations       map[string]string `json:"annotations"`
+	NamespaceSelector struct {
+		Any bool `json:"any"`
+	} `json:"namespaceSelector"`
+}
+
+type Duration time.Duration
+
+func (d Duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(d).String())
+}
+
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case string:
+		dd, err := time.ParseDuration(value)
+		if err != nil {
+			return err
+		}
+		*d = Duration(dd)
+		return nil
+	default:
+		return errors.New("invalid duration")
+	}
+}
+
+type Container struct {
+	SecurityContext struct {
+		AllowPrivilegeEscalation bool `json:"allowPrivilegeEscalation"`
+	} `json:"securityContext"`
+	Resources struct {
+		Request struct {
+			CPU    *resource.Quantity `json:"cpu"`
+			Memory *resource.Quantity `json:"memory"`
+		} `json:"request"`
+		Limits struct {
+			CPU    *resource.Quantity `json:"cpu"`
+			Memory *resource.Quantity `json:"memory"`
+		} `json:"limits"`
+		JavaMaxHeapSize *resource.Quantity `json:"javaMaxHeapSize"`
+	} `json:"resources"`
+	JavaGCLogEnabled string `json:"javaGCLogEnabled"` // XXX ugh - it ends up as an env var
+}
+
+type Deployment struct {
+	Create   bool `json:"create"`
+	Strategy struct {
+		Type string `json:"type"`
+	} `json:"strategy"`
+	SchedulerName  string `json:"schedulerName"`
+	UpdateStrategy struct {
+		Type string `json:"type"`
+	} `json:"updateStrategy"`
+	Budget struct {
+		MaxUnavailable int32 `json:"maxUnavailable"`
+	} `json:"budget"`
+	Annotations    map[string]string `json:"annotations"`
+	LivenessProbe  *corev1.Probe     `json:"livenessProbe,omitempty"`
+	ReadinessProbe *corev1.Probe     `json:"readinessProbe,omitempty"`
+	ExtraEnv       []struct {
+		Name  string `json:"name"`
+		Value string `json:"value"`
+	} `json:"extraEnv"`
+	ExtraEnvFrom []struct {
```

Comment:
nit: `[]corev1.EnvFromSource` (I think?) works here though it's a bit more permissive.